### PR TITLE
Cheese staff now properly reverts after a duration

### PR DESCRIPTION
**Pull request by Moltijoe**
Missatessatessy asked me to fix this

:cl:  
bugfix: cheese staff now properly uncheesifies
/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
you can also *suicide to immediately get out of it lmao

**Comment by Moltijoe** [comment]
> you can also *suicide to immediately get out of it lmao

ugh

**Comment by missatessatessy** [comment]
Cheese staff turning you back to human was always the intended function, I just was never a good enough coder to do it. I asked Molti to code this for me, so I approve these changes if they work.

**Comment by missatessatessy** [comment]
I played around with the code on a private server few issues: 
- For some reason in the middle of the transformation you drop what you had in your hands (not at the start and not at the end, in the middle while you're a cheesewheel).
- When uncheesify procs it leaves the person stunned and almost crits them, eventually they heal to full, but I don't want people to be this damaged and stunned when they come out.
- When uncheesify procs it needs to delete the cheese mob, currently the cheese mob dies, leaving its body and extra cheese on the ground.
- The message saying "you have fallen out" doesn't appear.

With my limited knowledge of code i think i get why 2 of these things are the way they are, but the other two i dunno.

File: code/modules/mob/living/simple_animal/friendly/cheese.dm [hunk 1]
```diff
@@ -67,3 +70,19 @@
 		uncheeseify(src)
 	. = ..()
 	
+/mob/living/simple_animal/cheese/proc/uncheeseify(mob/living/simple_animal/cheese/cheese)
+	if(cheese.stored_mob)
+		var/mob/living/L = cheese.stored_mob
+		var/mob/living/simple_animal/cheese/C = cheese
+		L.forceMove(get_turf(C))
+		C.stored_mob = null
+		to_chat(L, "<span class='big bold'>You have fallen out of the cheese wheel!</b>")
```
Comment (missatessatessy) [inline code/modules/mob/living/simple_animal/friendly/cheese.dm]: I think this needs to be placed after the mind transfer

File: code/modules/mob/living/simple_animal/friendly/cheese.dm [hunk 2]
```diff
@@ -67,3 +70,19 @@
...
+		if(L.mind)
+			C.mind.transfer_to(L)
+		else
+			L.key = C.key
+		C.transfer_observers_to(L)
+		C.death()
```
Comment (missatessatessy) [inline code/modules/mob/living/simple_animal/friendly/cheese.dm]: C.del would be preferred here ( i think, whatever removes the cheese mob)

File: code/modules/projectiles/projectile/magic.dm [hunk 3]
```diff
@@ -298,6 +298,9 @@
 	var/mob/living/simple_animal/cheese/B = new(M.loc)
 	if(!B)
 		return
+	M.dropItemToGround(M.get_active_held_item())
+	M.dropItemToGround(M.get_inactive_held_item())
```
Comment (ReddicusDragon) [inline code/modules/projectiles/projectile/magic.dm:302]: Does this just make the mob the cheese used to be drop the currently held items?

File: code/modules/mob/living/simple_animal/friendly/cheese.dm [hunk 4]
```diff
@@ -22,9 +22,14 @@
...
 	density = FALSE
 	var/mob/living/stored_mob
+	var/temporary = FALSE //permanent until made temporary
 
 /mob/living/simple_animal/cheese/Life()
 	..()
+	if(stored_mob)
+		stored_mob.life_tickrate = 0
+	if(temporary)
+		addtimer(CALLBACK(src, .proc/uncheeseify, src), 1 MINUTES, TIMER_UNIQUE)
```
Comment (ReddicusDragon) [inline code/modules/mob/living/simple_animal/friendly/cheese.dm:32]: I love being cheesed for 1 minute

File: code/modules/projectiles/projectile/magic.dm (hunk 3 above)
Comment (Moltijoe) [reply code/modules/projectiles/projectile/magic.dm:302]: yes

**Review by ReddicusDragon — APPROVED** [review]
I see nothing wrong here, uncheeseify seems good too. I trust Molti enough to test their PRs.

File: code/modules/mob/living/simple_animal/friendly/cheese.dm (hunk 4 above)
Comment (Moltijoe) [reply code/modules/mob/living/simple_animal/friendly/cheese.dm:32]: this was how it was previously, it just didn't actually work

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**